nominal/core/connection.py

```diff
@@ -141,13 +152,104 @@ def get_channel(self, name: str, tags: dict[str, str] | None = None) -> Channel:
         series = self._clients.logical_series.get_logical_series(self._clients.auth_header, resolved_series.rid)
         return Channel._from_conjure_logicalseries_api(self._clients, series)
 
-    def get_nominal_write_stream(self, batch_size: int = 50_000, max_wait_sec: int = 1) -> WriteStream:
-        """get_nominal_write_stream is deprecated and will be removed in a future version,
-        use get_write_stream instead.
+    def archive(self) -> None:
+        """Archive this connection.
+        Archived connections are not deleted, but are hidden from the UI.
+        """
+        self._clients.connection.archive_connection(self._clients.auth_header, self.rid)
+
+    def unarchive(self) -> None:
+        """Unarchive this connection, making it visible in the UI."""
+        self._clients.connection.unarchive_connection(self._clients.auth_header, self.rid)
+
+
+@dataclass(frozen=True)
+class NominalStreamingConnection(Connection):
+    nominal_data_source_rid: str
+
+    def process_batch(self, batch: Sequence[BatchItem], data_format: Literal["json", "protobuf"] = "json") -> None:
+        """Process a batch of items."""
+        if data_format == "json":
+            process_batch_legacy(
+                batch, self.nominal_data_source_rid, self._clients.auth_header, self._clients.storage_writer
+            )
+        else:
+            from nominal.core.batch_processor_proto import process_batch
+
+            process_batch(
+                batch=batch,
+                nominal_data_source_rid=self.nominal_data_source_rid,
+                auth_header=self._clients.auth_header,
+                proto_write=self._clients.proto_write,
+            )
+
+    def get_write_stream_v2(
+        self,
+        batch_size: int = 50_000,
+        max_wait: timedelta = timedelta(seconds=1),
+        data_format: Literal["json", "protobuf"] = "json",
+        backpressure_mode: BackpressureMode = BackpressureMode.BLOCK,
+        maxsize: int = 0,
+    ) -> WriteStreamV2:
+        """Stream to write non-blocking messages to a datasource.
+
+        Args:
+        ----
+            batch_size (int): How big the batch can get before writing to Nominal. Default 50,000
```

Comment:
nit: let's name this `max_batch_size` since it's more in line with `max_wait` (while we're here, "wait" is kinda a weird name for what it is)